### DISCUSSION TITLE
PR for align and justify arbitrary values AND :not() CSS pseudo-class ?

### DISCUSSION THREAD

**Discussion by paulm17** [tailwindlabs/tailwindcss#11067] [Ideas]
Would a PR be welcome so that arbitrary values for align and justify are supported?

```
align-[--align]
justify-[--justify]
```

Also using not for an arbitrary variant is annoying.  I have a use case where

```
[&:not(:last-of-type)::before]
```

Should really be:

```
not(:last-of-type):before:
```

Seeing as last-of-type and before are already supported!

I don't know if it's possible.  But I'm currently forking over mantine to support tailwind and doing bracket notation for these examples is annoying 😓.  Would like to see native support. 

Thanks

**Comment by lewis374** [comment]
Hello,
In general, open source projects typically welcome contributions from the community, including pull requests (PRs) that add new features or functionality.

If you have identified a need for additional features in a particular open source project, such as support for arbitrary values for align and justify, or an update to the syntax of existing selectors, you can create a pull request with your proposed changes.

Before submitting your PR, it's a good idea to check the project's documentation or guidelines for contributions to ensure that your changes meet their standards and requirements. You may also want to discuss your proposed changes with the project maintainers or other contributors to get their feedback and make sure that your changes are aligned with the project's goals and direction.

Overall, the open source community values collaboration and contributions, so if you have a suggestion or improvement to make, it's worth considering submitting a PR. hope so it is useful for you . [MyWaffleHouse.com](https://www.mywafflehouse.net/)

**Comment by paulm17** [comment]
@lewis374 

Smells like a GPT chat answer.  Please don't do this.

In the Pull requests section of: https://github.com/tailwindlabs/tailwindcss/blob/master/.github/CONTRIBUTING.md

> Please ask first before starting work on any significant new features.
> 
> It's never a fun experience to have your pull request declined after investing a lot of time and effort into a new feature. To avoid this > from happening, we request that contributors create [a feature request](https://github.com/tailwindlabs/tailwindcss/discussions/new?category=ideas) to first discuss any significant new ideas. This includes things like adding new utilities, creating new at-rules, etc.

Hence the question.
